feat: add ignore reversed record to vouchers

jniles opened the pull request:
Implements #5278.

In this case, we provide three options as suggested by @mbayopanda.

1. Exclude all reversed records
2. Include only reversed records
3. Exclude all reversed records _and_ reversal records

This allows users to filter out both the records that are reversed on the credit/debit notes that are made against those records.

Closes #5278.

![image](https://user-images.githubusercontent.com/896472/105463091-d810f200-5c8f-11eb-9a7f-4174e6dcea0b.png)

jniles committed:
feat(vouchers): server API for reversed filter

Implements and tests the server API for filtering on reversed vouchers.

jniles committed:
feat(vouchers): implement client-side

Implement client-side for reversed records filter for voucher registry.

Closes #5278.

mbayopanda commented on client/src/modules/vouchers/vouchers.service.js:
```diff
@@ -47,7 +47,7 @@ function VoucherService(
     { key : 'user_id', label : 'FORM.LABELS.USER' },
     { key : 'reference', label : 'FORM.LABELS.REFERENCE' },
-    { key : 'reversed', label : 'FORM.INFO.ANNULLED' },
+    { key : 'reversed', label : 'VOUCHERS.GLOBAL.REVERSED_RECORDS' },
```
👍🏽

mbayopanda reviewed (commented):
The feature is working fine, we have just a text not translated in the modal of reversal

![image](https://user-images.githubusercontent.com/5445251/104808664-069e5100-57e8-11eb-88cd-14ea021a96d9.png)

And what about the credit note row added after a reversal? We clean the registry by removing reserved voucher but we still have new lines added by the reversal operation

![image](https://user-images.githubusercontent.com/5445251/104808736-85938980-57e8-11eb-96ac-d874c9db4125.png)

jniles committed:
feat(voucher): improve reversal filter

Improves the reversal filter with a third option that allows you to
exclude both the reversed vouchers and the vouchers that will be
reversed in the query.

mbayopanda reviewed (approved):
Excellent 👍🏽 

The feature works as expected

bors r+
